### ISSUE TITLE
Releases should be created after all the artifacts are ready

### ISSUE THREAD

**Issue by swiatekm-sumo** [open-telemetry/opentelemetry-operator#2304]
### Component(s)

operator, target allocator, opamp bridge

### Describe the issue you're reporting

We create and publish the release before our container images are published. Normally, this makes no difference, but it's bitten us a bit due to #2295, and would also cause problems if any of the builds failed.

Unfortunately, fixing this isn't that easy, as Github Actions only has very basic support for dependencies between workflows. In order for this to happen, we'd need to refactor our CI into reusable workflows, and then have all the building and publishing happen in a dedicated release workflow.

A simple fix in the meantime would be to create the release in draft mode and have the release manager manually publish it once everything's fine. WDYT @jaronoff97 @pavolloffay ?

**Comment by jaronoff97** [comment]
Yeah i think we can just reorder the stages. A frustration i have with the current process is that we have to manually put the changelog in to the release notes. It would be great if tagging the release did all of that after all the images have built.
